backend/typescript/middlewares/validators/util.ts

```diff
@@ -23,12 +23,19 @@ export const validatePrimitive = (value: any, type: Type): boolean => {
     case "integer": {
       return typeof value === "number" && Number.isInteger(value);
     }
+    case "decimal": {
+      return typeof value === "number" && !Number.isNaN(value);
+    }
     default: {
       return false;
     }
   }
 };
 
+export const validateEnum = (value: any, enumType: any): boolean => {
```

Comment:
Nice! This is good to have, we should update the other validators to use this too in the future :)